starter_code/src/movies.js

```diff
@@ -1,12 +1,48 @@
 /* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
 
-
-// Get the average of all rates with 2 decimals 
-
+function ratesAverage(array) {
+
+    // Get the average of all rates with 2 decimals 
+    var accumulatedRates = 
+    array.reduce(function (acc, oneMovie) {
+        return acc + Number(oneMovie.rate);
+    }, 0);
+    
+    var averageRateNotRounded = accumulatedRates / array.length;
+    var averageRateRounded = Math.round(averageRateNotRounded *100)/100;
+    
+    return averageRateRounded;
+}
 
 // Get the average of Drama Movies
-
+function dramaMoviesRate (array) {
+    if (array.length === 1) {
+        var averageRateRounded = array[0].rate
+        return averageRateRounded;
+    }
+    else {
+        var dramaArray = 
+        array.filter(function (oneMovie){
+        return oneMovie.genre.indexOf('Drama') >= 0;
+        });
+        var accumulatedRates = 
+            dramaArray.reduce(function (acc, oneMovie) {
+                if (oneMovie.rate === undefined) { 
+                    return acc;
+                }
+                else {
+                return acc + Number(oneMovie.rate);}
+            }, 0);
+        
+        var averageRateNotRounded = accumulatedRates / dramaArray.length;
+        var averageRateRounded = Math.round(averageRateNotRounded *100)/100;
+        
+        return averageRateRounded;
+      }
+    }
```

Comment:
Good job, this works ! 
a few comments : 
- we now have access to a method "includes", so we dont really need anymore this little trick with "indexOf > 0" to check if something exists.
- Also,  to simplify your code and keep it dry, you can re-use the rateAvarage function above after you have filtered the array 
- to get rid of the last error on jasmine, you can wrap the code inside the function (or only the return) inside an if statement :   
if (dramaArray.length > 0) {//code here}